worker/worker.go

```diff
@@ -62,6 +82,52 @@ func HandleJoinResponse(response *utils.HttpResponse, w *Worker) {
 	w.QueueId = queueId
 }
 
+func (w *Worker) CheckAuthentication() error {
+	err := utils.CheckToken(w.Token)
```

Comment:
why are we checking the Token here? aren't we already checking before the GetTask call?

Reply:
YOu got a point, actually this function doesn't need to exist. I've removed it and checked the queueId inside the getTask method.